resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -518,9 +518,31 @@ private void followCname(InetSocketAddress nameServerAddr, String name, String c
             trace.append(cname);
         }
 
+        boolean queryARecord = false;
+        boolean queryAAAARecord = false;
+        for (InternetProtocolFamily family: resolveAddressTypes) {
```

Comment:
we iterate over `resolveAddressTypes` which is a `final` array and I don't think the intention is the contents will change. I wonder if we should predetermine these types of conditions and store them in `final` variables to avoid constantly iterating. For example having a member variables:

```java
private final boolean cnameFollowARecords;
private final boolean cnameFollowAAAARecords;
```

We can check the pre-condition that at least one must be set in the constructor too.